monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,541 @@
+            average: the operation is performed after mixing all modalities.
+            is_label: If the data is from label stats.
+        """
+        axis = None
+        if type(data[0]) is list or type(data[0]) is np.array:
+            if not is_label:
+                # data size = [num of cases, number of modalities, stats]
+                data = np.concatenate([[np.array(_) for _ in data]])
+            else:
+                # data size = [num of cases, stats]
+                data = np.concatenate([np.array(_) for _ in data])
+            axis = (0,)
+            if average and len(data.shape) > 2:
+                axis = (0, 1)
+        # Calculate statistics from the data using numpy. The torch max, min, median e.t.c have
+        # inconsistent interface for axis/dim input. Only used for summary
+        result = {}
+        for name, ops in self.operations_np.items():
+            # get results
+            _result = ops(np.array(data), axis=axis).tolist()  # post process with key mapping
+            mappingkeys = self.operations_mappingkey.get(name, None)
+            if mappingkeys is not None:
+                result.update({mappingkeys[i]: _result[i] for i in range(len(_result))})
+            else:
+                result[name] = _result
+        return result
+
+    def _stats_opt(self, data):
+        """Calculate statistics from the data"""
+        result = {}
+        for name, ops in self.operations.items():
+            if len(data) == 0:
+                data = torch.tensor([0.0], device=self.device)
+            if not torch.is_tensor(data):
+                data = torch.from_numpy(data).to(self.device)
+            #  compute the results
+            # torch.quantile may fail with large input, if failed, use numpy version
+            try:
+                _result = ops(data).data.cpu().numpy().tolist()
+            except Exception:
+                warnings.warn(f"torch {name} version has failed, using np version.")
+                _result = self.operations_np[name](data.cpu().numpy()).tolist()
+                pass
+            # post process the data
+            mappingkeys = self.operations_mappingkey.get(name, None)
+            if mappingkeys is not None:
+                result.update({mappingkeys[i]: _result[i] for i in range(len(_result))})
+            else:
+                result[name] = _result
+        return result
+
+    def _get_foreground_image(self, image, label=None):
+        """
+        Update select_fn if the foreground is defined differently.
+        return image foreground without using label
+
+        """
+        crop_foreground = monai.transforms.CropForeground(select_fn=lambda x: x > 0)
+        return crop_foreground(image)
+
+    def _get_foreground_label(self, image, label):
+        """Define foreground using foreground label"""
+        return image[label > 0]
+
+    def save_yaml(self, results):
```

Comment:
Can we use the MONAI API `export_config_file`?
https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/config_parser.py#L391

Thanks.

Reply:
Fixed in a1055d8